test/functional/forall/tests/test-forall-atomic-basic.hpp

```diff
@@ -0,0 +1,165 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Header file containing basic functional tests for atomic operations with forall.
+///
+
+#ifndef __TEST_FORALL_ATOMIC_BASIC_HPP__
+#define __TEST_FORALL_ATOMIC_BASIC_HPP__
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA_gtest.hpp"
+#include "test-forall-utils.hpp"
+
+// Type parameterized test for experimentation/discussion.
+TYPED_TEST_SUITE_P(SeqForallAtomicBasicFunctionalTest);
+
+template <typename T>
+class SeqForallAtomicBasicFunctionalTest : public ::testing::Test
+{
+};
+
+#if defined(RAJA_ENABLE_CUDA)
+TYPED_TEST_SUITE_P(CudaForallAtomicBasicFunctionalTest);
+
+template <typename T>
+class CudaForallAtomicBasicFunctionalTest : public ::testing::Test
+{
+};
+#endif
+
+#if defined(RAJA_ENABLE_HIP)
+TYPED_TEST_SUITE_P(HipForallAtomicBasicFunctionalTest);
+
+template <typename T>
+class HipForallAtomicBasicFunctionalTest : public ::testing::Test
+{
+};
+#endif
```

Comment:
I think we only need to use a single ForallAtomicBasicFunctionalTest class, the different backends are defined by their given instantiated tests with their typelists in the .cpp files

Reply:
but then again I see that you're calling the CUDA/HIP tests with GPU_TEST, which I don't know if we still need to be using ...

Reply:
Yes, defining a GPU_TYPED_* requires another registration/instantiation. I'm not sure if it's necessary either, but it's not difficult to revert.